azurelinuxagent/agent.py

```diff
@@ -235,15 +236,29 @@ def collect_logs(self, is_full_mode):
             # Running log collector resource monitoring only if agent starts the log collector.
             # If Log collector start by any other means, then it will not be monitored.
             if CollectLogsHandler.is_enabled_monitor_cgroups_check():
-                for metric in tracked_metrics:
-                    if isinstance(metric, CpuMetrics):
-                        metric.initialize_cpu_usage()
+                for controller in tracked_controllers:
+                    if isinstance(controller, _CpuController):
+                        controller.initialize_cpu_usage()
                         break
-                log_collector_monitor = get_log_collector_monitor_handler(tracked_metrics)
+                log_collector_monitor = get_log_collector_monitor_handler(tracked_controllers)
                 log_collector_monitor.run()
-            archive = log_collector.collect_logs_and_get_archive()
+
+            archive, total_uncompressed_size = log_collector.collect_logs_and_get_archive()
             logger.info("Log collection successfully completed. Archive can be found at {0} "
                   "and detailed log output can be found at {1}".format(archive, OUTPUT_RESULTS_FILE_PATH))
+
+            if log_collector_monitor is not None:
+                log_collector_monitor.stop()
+                try:
+                    msg = "Resource usage summary: total uncompressed file size={0}; {1}".format(
+                        total_uncompressed_size, log_collector_monitor.get_metrics_summary())
+                    logger.info(msg)
+                    event.add_event(op=event.WALAEventOperation.LogCollection, message=msg, log_event=False)
```

Comment:
I have split thought about whether summary is useful. This msg bunch of information, someone has to parse this msg to understand different metrics across the fleet. Do we already not capture as part of monitor thread?

Reply:
We don't capture log collector metrics in telemetry right now. I'm not sure how we would determine if these new limits are appropriate without sending events for these metrics

Reply:
As part of LogCollectorMonitorHandler thread, when we poll for metrics, we send to telemetry as well. I think you used same data to evaluate limits. Now logging same data in different table is not useful I feel


```
                    metrics = self._poll_resource_usage()
                    self._send_telemetry(metrics)
                    self._verify_memory_limit(metrics)
```

Reply:
How often is this telemetry sent? the log collector process is short in duration. if the polling period is too long, we won't capture enough info; if it is too short, we may be flooding telemetry. A summary seems appropriate.

As for the parsing, should we send this as JSON instead?